compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -475,6 +476,28 @@ convertSimpleRecordDef env tycon = do
 
     pure $ typeDef : [workerDef | flavour == NewtypeFlavour]
 
+convertTypeSynonym :: Env -> TyCon -> ConvertM [Definition]
+convertTypeSynonym env tycon
+    | NameIn DA_Generics _ <- GHC.tyConName tycon
+    = pure []
+
+    | envLfVersion env `supports` featureTypeSynonyms
+    , Just (params, body) <- synTyConDefn_maybe tycon
+    , isLiftedTypeKind (tyConResKind tycon)
```

Comment:
A brief comment on this check and the one below would be great. This one seems to be to get `*`, `Constraint`, …. I’m not quite sure about the second one.